File: HW#01/src/Unal_Baris.py

#****************************************************************************

#CS315 HW01 - ASSOCIATIVE ARRAYS - PYTHON
#Barış Tan Ünal
#22003617
#Section: 03

#****************************************************************************


# 1. Initialize
print( "\n\n", "1. Initialize" );
books = {
  "Frankenstein": { "author": "Mary Shelley", "availability": True },
  "Utopia": { "author": "Thomas More", "availability": False },
  "The Illiad": { "author": "Homer", "availability": True },
  "The Tempest": { "author": "Shakespeare", "availability": False }
}
for x in books:
    print( "\n", x )
    for y in books[x]:
        print( y, ":" ,books[x][y] )

# 2. Get the value for a given key
print( "\n\n", "2. Get the value for a given key" );
print( books[ "Utopia" ] );

# 3. Add a new element
print( "\n\n", "3. Add a new element" );
books[ "The Strange Case of Dr. Jekyll and Mr. Hyde" ] = { "author": "R. L. Stevenson", "availability": False  }
for x in books:
    print( "\n", x )
    for y in books[x]:
        print( y, ":" ,books[x][y] )

# 4. Remove an element
print( "\n\n", "4. Remove an element" );
del books[ "The Tempest" ];
for x in books:
    print( "\n", x )
    for y in books[x]:
        print( y, ":" ,books[x][y] )

# 5. Modify the value of an existing element
print( "\n\n", "5. Modify the value of an existing element" );
books[ "Frankenstein" ] = { "author": "Mary Shelley", "availability": False };
for x in books:
    print( "\n", x )
    for y in books[x]:
        print( y, ":" ,books[x][y] )

# 6. Search for the existence of a key
print( "\n\n", "6. Search for the existence of a key" );
def checkKey( dict, key ):
    if key in dict.keys():
        print( key, " exists.")
    else:
        print( key, " does not exist.")
checkKey( books, "1984" )
checkKey( books, "Frankenstein" )

# 7. Search for the existence of a value
print( "\n\n", "7. Search for the existence of a value" )
def checkValue( dict, value ):
    if value in dict.values():
        print( value, " exists.")
    else:
        print( value, " does not exist.")
checkValue( books, { "author": "Mary Shelley", "availability": True } )
checkValue( books, { "author": "Homer", "availability": True }, )

# 8. Loop through an associative array, apply a function, called foo, which simply prints the key-value pair
print( "\n\n", "8. Loop through an associative array, apply a function, called foo, which simply prints the key-value pair" )
def foo( dict ):
  for i in dict:
    print( "\n", i )
    for j in dict[i]:
        print( j, ":" ,dict[i][j] )

foo( books )

File: HW#01/src/Unal_Baris.rs

/*
****************************************************************************

CS315 HW01 - ASSOCIATIVE ARRAYS - RUST
Barış Tan Ünal
22003617
Section: 03

****************************************************************************
*/

use std::collections::HashMap;

fn main() {
    println!("Euroleage Players Associative Array");
    
    // 1. Initialize
    let mut euroleague_players = HashMap::new();
    
    euroleague_players.insert(
        "Marko Guduric", "Fenerbahce"
    );
    euroleague_players.insert(
        "Mike James", "Monaco"
    );
    euroleague_players.insert(
        "Walter Tavares", "Real Madrid"
    );
    
    // 2. Get the value for a given key
    println!("Marko Guduric plays for: {}", euroleague_players["Marko Guduric"]);
    
    // 3. Add a new element
    euroleague_players.insert( "Will Clyburn", "CSKA Moscow" );
    
    // 4. Remove an element
    euroleague_players.remove( "Walter Tavares" );
    
     for (key, value) in &euroleague_players {
        println!( "{}: {}", key, value );
    }
    
    // 5. Modify the value of an existing element
    euroleague_players.insert( "Will Clyburn", "Anadolu Efes" );
    
    for (key, value) in &euroleague_players {
        println!( "{}: {}", key, value );
    }
    
    // 6. Search for the existence of a key
    if euroleague_players.contains_key("Mike James") {
        println!("Mike James plays for a team in Euroleague.");
    }
    
    // 7. Search for the existence of a value
    let team_to_find = "Fenerbahce";
    for (_player, team) in &euroleague_players {
        if *team == team_to_find {
            println!("There is a player that plays for Fenerbahce.");
        }
    }
    
    
    // 8. Loop through an associative array, apply a function, called foo, which simply prints the key-value pair
    
    fn foo ( arr: &mut HashMap<&str, &str> ) {
        for ( player, team ) in arr {ß
            println!( "{}: {}", player, team );
        }
    }
    
    foo ( &mut euroleague_players);
}